src/TelemetryChannels/ServerTelemetryChannel/Shared/Implementation/PartialSuccessTransmissionPolicy.cs

```diff
@@ -0,0 +1,153 @@
+                    
+                    byte[] data = JsonSerializer.ConvertToByteArray(newTransmissions);
+                    Transmission newTransmission = new Transmission(
+                        args.Transmission.EndpointAddress,
+                        data,
+                        args.Transmission.ContentType,
+                        args.Transmission.ContentEncoding,
+                        args.Transmission.Timeout);
+
+                    this.Transmitter.Enqueue(newTransmission);
+                }
+            }
+        }
+
+        private string ParsePartialSuccessResponse(Transmission initialTransmission, TransmissionProcessedEventArgs args)
+        {
+            BackendResponse backendResponse;
+            string responseContent = args.Response.Content;
+            try
+            {
+                backendResponse = this.serializer.Deserialize<BackendResponse>(responseContent);
+            }
+            catch (ArgumentException exp)
+            {
+                TelemetryChannelEventSource.Log.BreezeResponseWasNotParsedWarning(exp.Message, responseContent);
+                this.ConsecutiveErrors = 0;
+                return null;
+            }
+            catch (InvalidOperationException exp)
+            {
+                TelemetryChannelEventSource.Log.BreezeResponseWasNotParsedWarning(exp.Message, responseContent);
+                this.ConsecutiveErrors = 0;
+                return null;
+            }
+
+            string newTransmissions = null;
+
+            if (backendResponse != null && backendResponse.ItemsAccepted != backendResponse.ItemsReceived)
+            {
+                string[] items = JsonSerializer
+                    .Deserialize(initialTransmission.Content)
+                    .Split(new[] { Environment.NewLine }, StringSplitOptions.RemoveEmptyEntries);
+
+                foreach (var error in backendResponse.Errors)
+                {
+                    if (error != null)
+                    {
+                        if (error.Index >= items.Length || error.Index < 0)
+                        {
+                            TelemetryChannelEventSource.Log.UnexpectedBreezeResponseWarning(items.Length, error.Index);
+                            continue;
+                        }
+
+                        TelemetryChannelEventSource.Log.ItemRejectedByEndpointWarning(error.Message);
+
+                        if (error.StatusCode == ResponseStatusCodes.RequestTimeout ||
+                            error.StatusCode == ResponseStatusCodes.ServiceUnavailable ||
+                            error.StatusCode == ResponseStatusCodes.InternalServerError ||
+                            error.StatusCode == ResponseStatusCodes.ResponseCodeTooManyRequests ||
+                            error.StatusCode == ResponseStatusCodes.ResponseCodeTooManyRequestsOverExtendedTime)
+                        {
+                            if (string.IsNullOrEmpty(newTransmissions))
```

Comment:
Looks like you are not initializing newTransmissions anywhere so it will always remain null. Did you forget to call ParsePartialSuccessResponse?